### PR TITLE
v1.11 Backports 2023-05-03

### PR DESCRIPTION
**Pull request by joamaki**
* [ ] #25043 (@harsimran-pabla)
 * [ ] #25046 (@nbusseneau)
 * [ ] #24770 (@aditighag)
 * [x] #25033 (@giorio94)
 * [x] #25109 (@sayboras)
 * [x] #25192 (@kaworu)
 * [ ] #25174 (@joestringer)
 * [x] #25219 (@tommyp1ckles)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 25043 25046 24770 25033 25109 25192 25174 25219; do contrib/backporting/set-labels.py $pr done 1.11; done
```
or with
```
make add-labels BRANCH=v1.11 ISSUES=25043,25046,24770,25033,25109,25192,25174,25219
```

### CODE REVIEW

**Review by giorio94 — APPROVED** [review]
My commit looks good. Thanks!

**Comment by harsimran-pabla** [comment]
BGP change https://github.com/cilium/cilium/pull/25043 is already backported in this https://github.com/cilium/cilium/pull/25139

**Review by kaworu — APPROVED** [review]
My commits LGTM, thanks @joamaki!

**Review by sayboras — APPROVED** [review]
Looks good for my commit, thanks 💯

**Review by aditighag — APPROVED** [review]
Looks good for my commits. Thanks!

**Review by tommyp1ckles — APPROVED** [review]
thanks1

**Comment by joamaki** [comment]
> Looks good for my commits. Thanks!

Hey, looks like I forgot to fix up pkg/service/service_test.go. After fixing it the same way as in the v1.12 backports I get a test failure:

```
FAIL: service_test.go:833: ManagerTestSuite.TestRestoreServiceWithTerminatingBackends

service_test.go:864:
    // Backends including terminating ones have been restored
    c.Assert(len(m.svc.backendByHash), Equals, 3)
... obtained int = 2
... expected int = 3

OOPS: 0 passed, 1 FAILED
```

Enabling debug logging I get:
```
level=debug msg="Upserting service" backends="[{0  {10.0.0.1 {TCP 8080} 0} false} {0  {10.0.0.2 {TCP 8080} 0} false} {0  {10.0.0.4 {TCP 8080} 0} false}]" loadBalancerSourceRanges="[]" serviceIP="{1.1.1.1 {TCP 80} 0}" serviceName= serviceNamespace= sessionAffinity=true sessionAffinityTimeout=100 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=NodePort
level=debug msg="Resolving service" l3n4Addr="{IP:1.1.1.1 L4Addr:{Protocol:TCP Port:80} Scope:0}" subsys=service
level=debug msg="Acquired service ID" backends="[{0  {10.0.0.1 {TCP 8080} 0} false} {0  {10.0.0.2 {TCP 8080} 0} false} {0  {10.0.0.4 {TCP 8080} 0} false}]" loadBalancerSourceRanges="[]" serviceID=1 serviceIP="{1.1.1.1 {TCP 80} 0}" serviceName= serviceNamespace= sessionAffinity=true sessionAffinityTimeout=100 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=NodePort
level=debug msg="Deleting backends from session affinity match" backends="[]" serviceID=1 subsys=service
level=debug msg="Adding new backend" backendID=1 backends="[{0  {10.0.0.1 {TCP 8080} 0} false} {0  {10.0.0.2 {TCP 8080} 0} false} {0  {10.0.0.4 {TCP 8080} 0} false}]" l3n4Addr="{10.0.0.1 {TCP 8080} 0}" loadBalancerSourceRanges="[]" serviceID=1 serviceIP="{1.1.1.1 {TCP 80} 0}" serviceName= serviceNamespace= sessionAffinity=true sessionAffinityTimeout=100 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=NodePort
level=debug msg="Adding new backend" backendID=2 backends="[{0  {10.0.0.1 {TCP 8080} 0} false} {0  {10.0.0.2 {TCP 8080} 0} false} {0  {10.0.0.4 {TCP 8080} 0} false}]" l3n4Addr="{10.0.0.2 {TCP 8080} 0}" loadBalancerSourceRanges="[]" serviceID=1 serviceIP="{1.1.1.1 {TCP 80} 0}" serviceName= serviceNamespace= sessionAffinity=true sessionAffinityTimeout=100 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=NodePort
level=debug msg="Adding new backend" backendID=3 backends="[{0  {10.0.0.1 {TCP 8080} 0} false} {0  {10.0.0.2 {TCP 8080} 0} false} {0  {10.0.0.4 {TCP 8080} 0} false}]" l3n4Addr="{10.0.0.4 {TCP 8080} 0}" loadBalancerSourceRanges="[]" serviceID=1 serviceIP="{1.1.1.1 {TCP 80} 0}" serviceName= serviceNamespace= sessionAffinity=true sessionAffinityTimeout=100 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=NodePort
level=debug msg="Adding backends to affinity match map" backends="[1 2 3]" serviceID=1 subsys=service
level=debug msg="Upserting service" backends="[{0  {10.0.0.1 {TCP 8080} 0} false} {0  {10.0.0.2 {TCP 8080} 0} false} {0  {10.0.0.4 {TCP 8080} 0} true}]" loadBalancerSourceRanges="[]" serviceIP="{1.1.1.1 {TCP 80} 0}" serviceName= serviceNamespace= sessionAffinity=true sessionAffinityTimeout=100 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=NodePort
level=debug msg="Acquired service ID" backends="[{0  {10.0.0.1 {TCP 8080} 0} false} {0  {10.0.0.2 {TCP 8080} 0} false} {0  {10.0.0.4 {TCP 8080} 0} true}]" loadBalancerSourceRanges="[]" serviceID=1 serviceIP="{1.1.1.1 {TCP 80} 0}" serviceName= serviceNamespace= sessionAffinity=true sessionAffinityTimeout=100 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=NodePort
level=debug msg="Deleting backends from session affinity match" backends="[]" serviceID=1 subsys=service
level=debug msg="Adding backends to affinity match map" backends="[1 2 3]" serviceID=1 subsys=service
level=warning msg="Unable to add entry to affinity match map" backendID=1 error="Backend 1 already exists in 1 affinity map" serviceID=1 subsys=service
level=warning msg="Unable to add entry to affinity match map" backendID=2 error="Backend 2 already exists in 1 affinity map" serviceID=1 subsys=service
level=warning msg="Unable to add entry to affinity match map" backendID=3 error="Backend 3 already exists in 1 affinity map" serviceID=1 subsys=service
level=debug msg="Restoring service" serviceID=1 serviceIP="1.1.1.1:80" subsys=service
level=debug msg="Restoring service" l3n4Addr="{IP:1.1.1.1 L4Addr:{Protocol:NONE Port:80} Scope:0}" subsys=service
level=info msg="Restored services from maps" failedServices=0 restoredServices=1 subsys=service
level=debug msg="Restoring backend" backendID=1 l3n4Addr="10.0.0.1:8080" subsys=service
level=debug msg="Restoring backend" backendID=2 l3n4Addr="10.0.0.2:8080" subsys=service
level=debug msg="Restoring backend" backendID=3 l3n4Addr="10.0.0.4:8080" subsys=service
level=debug msg="Leaked backend entry not restored" backendID=3 l3n4Addr="{10.0.0.4 {NONE 8080} 0}" subsys=service
level=info msg="Restored backends from maps" failedBackends=0 restoredBackends=2 skippedBackends=1 subsys=service
```

@aditighag Any idea what's going on? Are we missing something important in v1.11?

File: pkg/service/service_test.go [hunk 1]
```diff
@@ -832,11 +835,16 @@ func (m *ManagerTestSuite) TestRestoreServiceWithTerminatingBackends(c *C) {
...
 		Type:                      lb.SVCTypeNodePort,
 		TrafficPolicy:             lb.SVCTrafficPolicyCluster,
+		SessionAffinity:           true,
+		SessionAffinityTimeoutSec: 100,
+		HealthCheckNodePort:       0,
+		Name:                      "",
+		Namespace:                 "",
+		LoadBalancerSourceRanges:  []*cidr.CIDR{},
 	}
+
```
Comment (aditighag) [inline pkg/service/service_test.go:847]: Can you revert these changes? Looks like there were accidentally added in the backport commit?

File: pkg/service/service_test.go [hunk 2]
```diff
@@ -904,3 +912,49 @@ func (m *ManagerTestSuite) TestDeleteServiceWithTerminatingBackends(c *C) {
...
 }
+
+func (m *ManagerTestSuite) TestRestoreServicesWithLeakedBackends(c *C) {
+	backends := backends1
+	p1 := &lb.SVC{
+		Frontend:  frontend1,
+		Backends:  backends,
+		Type:      lb.SVCTypeClusterIP,
+		Name:      "svc1",
+		Namespace: "ns1",
```
Comment (aditighag) [inline pkg/service/service_test.go:923]: This should be the only difference between 1.11 and newer branches.

**Review by aditighag — CHANGES_REQUESTED** [review]
Sorry, missed the differences around service name +namespace changes.

**Comment by YutaroHayakawa** [comment]
Since this branch comes from Jussi's fork, I cannot edit commits to apply Aditi's change. I'll move this to another PR the branch owned by me.

**Pull request closed without merging**